### ISSUE TITLE
Profile hero fields are flashing

### ISSUE THREAD

**Issue by vidvidvid** [MetaFam/TheGame#984]
### What happened?

All of the profile hero items are flashing and availability is not even displayed. Happens both locally and in production. Probably the fields are constantly being rerendered for some reason.

### What did you expect to happen?

The profile hero data should not be flashing :P

### How can we reproduce the problem (as minimally as possible)?

go to http://localhost:3000/player/vid
or https://my.metagame.wtf/player/vid

try other players as well

### Is there anything else we need to know?

related pr: https://github.com/MetaFam/TheGame/pull/963

### Assets

![flashing](https://user-images.githubusercontent.com/35112939/146820134-c5cdd17f-5b75-467e-bd25-d11769288fc0.gif)

**Comment by vidvidvid** [comment]
@firattale maybe you could look into it?

**Comment by vidvidvid** [comment]
Yup, did a little bit of logging and it seems like the component is rerendering like there's no tomorrow. Some vicious loop is going on.

**Comment by vidvidvid** [comment]
Sorry @firattale  😅 I went ahead and fixed the `useAnimation` hook that you made and implemented it in other profile sections. I just compared the old and new values and didn't retrigger the animation if they were the same. Somehow the `depends` value from useEffect was always different. Probably because
1. the `depends` variable was set
2. it triggered the `useAnimation`
3. the animation triggered the rendering of the component
4. the `depends` variable was again set and even though its value might have been the same, its reference wasn't, so this loop kept on repeating.
Not sure if this is the right interpretation, but I think what happened was probably something along these lines.
PR:
